lib/fileio/commands/baseExportCommand.cpp

```diff
@@ -0,0 +1,328 @@
+                   UsdMayaJobExportArgsTokens->defaultCameras.GetText(),
+                   MSyntax::kNoArg);
+    syntax.addFlag(kRenderLayerModeFlag,
+                   UsdMayaJobExportArgsTokens->renderLayerMode.GetText(),
+                   MSyntax::kString);
+    syntax.addFlag(kKindFlag,
+                   UsdMayaJobExportArgsTokens->kind.GetText(),
+                   MSyntax::kString);
+    syntax.addFlag(kCompatibilityFlag,
+                   UsdMayaJobExportArgsTokens->compatibility.GetText(),
+                   MSyntax::kString);
+
+    syntax.addFlag(kChaserFlag,
+                   UsdMayaJobExportArgsTokens->chaser.GetText(),
+                   MSyntax::kString);
+    syntax.makeFlagMultiUse(UsdMayaJobExportArgsTokens->chaser.GetText());
+
+    syntax.addFlag(kChaserArgsFlag,
+                   UsdMayaJobExportArgsTokens->chaserArgs.GetText(),
+                   MSyntax::kString, MSyntax::kString, MSyntax::kString);
+    syntax.makeFlagMultiUse(UsdMayaJobExportArgsTokens->chaserArgs.GetText());
+
+    syntax.addFlag(kMelPerFrameCallbackFlag,
+                   UsdMayaJobExportArgsTokens->melPerFrameCallback.GetText(),
+                   MSyntax::kNoArg);
+    syntax.addFlag(kMelPostCallbackFlag,
+                   UsdMayaJobExportArgsTokens->melPostCallback.GetText(),
+                   MSyntax::kNoArg);
+    syntax.addFlag(kPythonPerFrameCallbackFlag,
+                   UsdMayaJobExportArgsTokens->pythonPerFrameCallback.GetText(),
+                   MSyntax::kString);
+    syntax.addFlag(kPythonPostCallbackFlag,
+                   UsdMayaJobExportArgsTokens->pythonPostCallback.GetText(),
+                   MSyntax::kString);
+    syntax.addFlag(kVerboseFlag,
+                   UsdMayaJobExportArgsTokens->verbose.GetText(),
+                   MSyntax::kNoArg);
+
+    // These are additional flags under our control.
+    syntax.addFlag(kFrameRangeFlag, kFrameRangeFlagLong, MSyntax::kDouble, MSyntax::kDouble);
+    syntax.addFlag(kFrameStrideFlag, kFrameStrideFlagLong, MSyntax::kDouble);
+    syntax.addFlag(kFrameSampleFlag, kFrameSampleFlagLong, MSyntax::kDouble);
+    syntax.makeFlagMultiUse(kFrameSampleFlag);
+
+    syntax.addFlag(kAppendFlag, kAppendFlagLong, MSyntax::kBoolean);
+    syntax.addFlag(kFileFlag, kFileFlagLong, MSyntax::kString);
+    syntax.addFlag(kSelectionFlag, kSelectionFlagLong, MSyntax::kNoArg);
+
+    syntax.addFlag(kFilterTypesFlag, kFilterTypesFlagLong, MSyntax::kString);
+    syntax.makeFlagMultiUse(kFilterTypesFlag);
+
+    syntax.enableQuery(false);
+    syntax.enableEdit(false);
+
+    syntax.setObjectType(MSyntax::kSelectionList);
+    syntax.setMinObjects(0);
+
+    return syntax;
+}
+
+void* MayaUSDExportCommand::creator()
+{
+    return new MayaUSDExportCommand();
+}
+
+/* virtual */
+std::unique_ptr<UsdMaya_WriteJob> MayaUSDExportCommand::initializeWriteJob(const pxr::UsdMayaJobExportArgs & args)
+{
+    return std::unique_ptr<UsdMaya_WriteJob>(new UsdMaya_WriteJob(args));
```

Comment:
Both the Pixar plugin and the new MayaUSD plugin use UsdMaya_WriteJob.  This virtual function is here to allow derived plugins to change that behavior, but right now they are still identical.